src/controls/PolyControl.ts

```diff
@@ -0,0 +1,165 @@
+import { Point } from '../point.class';
+import { Control } from './control.class';
+import { TMat2D } from '../typedefs';
+import { iMatrix } from '../constants';
+import type { Polyline } from '../shapes/polyline.class';
+import { multiplyTransformMatrices } from '../util/misc/matrix';
+import { TPointerEvent, Transform } from '../EventTypeDefs';
+import { getLocalPoint } from './util';
+import { degreesToRadians } from '../util/misc/radiansDegreesConversion';
+import { TransformActionHandler } from '../EventTypeDefs';
+
+export class PolyControl extends Control {
+  pointIndex: number;
+  actionName = 'modifyPolygon';
+
+  constructor(options: Partial<Control>, pointIndex: number) {
+    super(options);
+    this.pointIndex = pointIndex;
+  }
+
+  static applySkew(point: Point, shear: Point) {
+    const skewedPoint = new Point();
+    skewedPoint.y = point.y + point.x * shear.y;
+    skewedPoint.x = point.x + skewedPoint.y * shear.x;
+    return skewedPoint;
+  }
+
+  static removeSkew(point: Point, shear: Point) {
+    const unskewedPoint = new Point();
+    unskewedPoint.x = point.x - point.y * shear.x;
+    unskewedPoint.y = point.y - unskewedPoint.x * shear.y;
+    return unskewedPoint;
+  }
+
+  static getSize(poly: Polyline) {
+    return new Point(poly.width, poly.height);
+  }
+
+  /**
+   * This function locates the controls.
+   * It'll be used both for drawing and for interaction.
+   */
+  positionHandler(dim: Point, finalMatrix: TMat2D, polyObject: Polyline) {
```

Comment:
is there a reason why finalMatrix is not good enough?

Reply:
I agree, finalMatrix should be used since it accounts for group

Reply:
I didn't understand very well what `finalMatrix` represents (only considers the `angle`of the object(?) ), but it is not the multiplication of the transformation matrix of the viewport and the object (what we need to correctly render the position of the control points)

Source of `finalMatrix`:

https://github.com/fabricjs/fabric.js/blob/b6f076cc0c817cd7f70abc7028c9d04295d121a8/src/shapes/Object/InteractiveObject.ts#L170-L197

Reply:
It transforms to the center of the target.
`dim` is the size in canvas plane.
But you are right, using it doesn't make sense in this context